code/modules/clothing/suits/armor.dm

```diff
@@ -70,6 +70,7 @@
 		for(var/X in actions)
 			var/datum/action/A = X
 			A.Remove(user)
+			actions -= X
```

Comment:
```suggestion
		QDEL_LIST_CONTENTS(actions)
```
Since the code is a bit oddly set up. You create an action each time you attach the holobadge to the suit. This action also needs to be deleted if you plan on creating a new one when attaching it again.
Doing a qdel will also remove the owner etc